src/megaapi_impl.cpp

```diff
@@ -11578,6 +11587,65 @@ void MegaApiImpl::chatlinkjoin_result(error e)
     fireOnRequestFinish(request, megaError);
 }
 
+void MegaApiImpl::folderlinkinfo_result(error e, handle owner, handle ph, string *attr, string* key, m_off_t currentSize, uint32_t numFiles, uint32_t numFolders, m_off_t versionsSize, uint32_t numVersions)
+{
+    MegaError megaError(e);
+    if (requestMap.find(client->restag) == requestMap.end()) return;
+    MegaRequestPrivate* request = requestMap.at(client->restag);
+    if (!request || (request->getType() != MegaRequest::TYPE_PUBLIC_LINK_INFORMATION)) return;
+
+    if (e == API_OK)
+    {
+        handle h;
+        size_t t = 0;
+        string nodekey;
+        string attrs;
+        byte* buf = NULL;
+        const char* k = NULL;
+        unsigned int keylength = FOLDERNODEKEYLENGTH;
+        byte binkey[FILENODEKEYLENGTH];
+        SymmCipher cipher;
+        SymmCipher* sc = &client->key;
+
+        if (key->size())
+        {
+            // Discard user handle for received key
+            while ((t = key->find_first_of(':', t)) != string::npos)
+            {
+               k = key->c_str() + (++t);
+               break;
+            }
+
+            // Decrypt the received key with our folder key
+            if (client->decryptkey(k, binkey, keylength, sc, 0, ph))
+            {
+                nodekey.assign((const char*)binkey, keylength);
+                if (cipher.setkey(&nodekey))
+                {
+                    // Decrypt attrs with the decrypted key
+                    buf = Node::decryptattr(&cipher, attr->c_str(), attr->size());
+                }
+            }
+        }
+        else
+        {
+            LOG_warn << "The folder link information doesn't contains the decryption key";
+            return fireOnRequestFinish(request, API_EINTERNAL);
+        }
+
+        MegaFolderInfoPrivate *folderInfo = new MegaFolderInfoPrivate(numFiles, numFolders - 1, numVersions, currentSize, versionsSize);
+        request->setNodeHandle(ph);
+        request->setParentHandle(owner);
+        request->setText((const char*)buf);
```

Comment:
Documentation says the `MegaRequest::getText()` will return the name of the folder, but here you don't get rid of the rest of attributes nor the prefix of each.